src/kv/msgpack_serialise.h

```diff
@@ -102,14 +93,34 @@ namespace kv
 
     kv::serialisers::SerialisedEntry read_next_pre_serialised()
     {
-#if MSGPACK_DONT_REPACK
+      // Check there are sufficient bytes for a size-prefix
+      constexpr auto size_size = sizeof(size_t);
```

Comment:
serialized::read() does this nicely.

Reply:
Sure, I've replaced the size-prefix parsing code with `serialized::read`. We need a little wrapping code around this, since this `MsgPackReader` is storing an incrementing `data_offset`, which has to be adjusted to match the decrementing `remainder` we pass to `serialized::read`. Hope the distinction is clear.